scripts/errcheck.sh

```diff
@@ -0,0 +1,28 @@
+# Check gofmt
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+echo "==> Checking for unchecked errors..."
+
+if ! which errcheck > /dev/null; then
+    echo "==> Installing errcheck..."
+    go get -u github.com/kisielk/errcheck
```

Comment:
Using `go get` like this will install errcheck as a dependency in the current go module (the provider) and will also modify the provider's `go.mod` file in the process.

Instead, use `go install github.com/kisielk/errcheck@latest`. This will install the binary under $GOPATH/bin and not affect the provider Go module. It's important here to have the version designator in place (`@latest`), otherwise it will still get installed as a provider dependency.

Check out `go help install` for a lot more detailed explanation.